airflow/utils.py

```diff
@@ -617,11 +619,20 @@ def handle_timeout(self, signum, frame):
         raise AirflowTaskTimeout(self.error_message)
 
     def __enter__(self):
-        signal.signal(signal.SIGALRM, self.handle_timeout)
-        signal.alarm(self.seconds)
+        try:
+            signal.signal(signal.SIGALRM, self.handle_timeout)
+            signal.alarm(self.seconds)
+        except Exception as e:
+            logging.warning(
+                "timeout could not be activated, it doens't work within "
+                "threads. Most likely you are using a LocalExecutor. "
+                "Timeouts aren't compatible with LocalExecutor")
 
     def __exit__(self, type, value, traceback):
-        signal.alarm(0)
+        try:
+            signal.alarm(0)
+        except Exception as e:
```

Comment:
Might be a good idea to log the exception (and maybe even the trace) too, even if it's only debug level